tests/components/enphase_envoy/test_sensor.py

```diff
@@ -0,0 +1,169 @@
+    ("sensor.envoy_1234_energy_consumption_today_l3", 3323 / 1000),
+    ("sensor.envoy_1234_energy_consumption_last_seven_days_l3", 3321 / 1000),
+    ("sensor.envoy_1234_lifetime_energy_consumption_l3", 3322 / 1000000),
+    ("sensor.inverter_1", 1),
+    ("sensor.envoy_1234_current_net_power_consumption", 101 / 1000),
+    ("sensor.envoy_1234_current_net_power_consumption_l1", 21 / 1000),
+    ("sensor.envoy_1234_current_net_power_consumption_l2", 31 / 1000),
+    ("sensor.envoy_1234_current_net_power_consumption_l3", 51 / 1000),
+    ("sensor.envoy_1234_lifetime_net_energy_consumption", 21234 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_consumption_l1", 212341 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_consumption_l2", 212342 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_consumption_l3", 212343 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_production", 22345 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_production_l1", 223451 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_production_l2", 223452 / 1000000),
+    ("sensor.envoy_1234_lifetime_net_energy_production_l3", 223453 / 1000000),
+]
+
+# list of sensor,state added as disabled, only test for existence
+DISABLED_TEST_ENTITIES = [
+    ("sensor.inverter_1_last_reported", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_metering_status_production_ct", 0),
+    ("sensor.envoy_1234_metering_status_production_ct_l1", 0),
+    ("sensor.envoy_1234_metering_status_production_ct_l2", 0),
+    ("sensor.envoy_1234_metering_status_production_ct_l3", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct_l1", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct_l2", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct_l3", 0),
+]
```

Comment:
I think you can avoid hard coding these, and fetch which entities are registered via the registry and use a snapshot assertion instead

Reply:
Yes. They assisted with initial value verification to make first snapshot had correct values.